providers/implementations/kdfs/scrypt.c

```diff
@@ -92,6 +94,38 @@ static void kdf_scrypt_reset(void *vctx)
     kdf_scrypt_init(ctx);
 }
 
+static void *kdf_scrypt_dup(void *vctx)
+{
+    const KDF_SCRYPT *src = (const KDF_SCRYPT *)vctx;
+    KDF_SCRYPT *dest;
+
+    dest = kdf_scrypt_new(src->libctx);
+    if (dest != NULL) {
+        if (src->sha256 != NULL && !EVP_MD_up_ref(src->sha256))
```

Comment:
Why does this one not use ossl_prov_digest_copy()?

Reply:
Because it's an EVP_MD not a PROV_DIGEST?